### PR TITLE
48109 Analytics and cookie management

### PR DESCRIPTION
**Pull request by Ashish15251**
Changes related to google analytics &  cookie management.

### CODE REVIEW

**Review by usha13529 — CHANGES_REQUESTED** [review]
Please fix the failing unit test cases.

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/appsettings.Development.json [hunk 1]
```diff
@@ -16,9 +16,10 @@
...
         "ConnectionString": ""
     },
-    "AzureAdB2C": {        
+    "AzureAdB2C": {
         "Instance": "",
         "ClientId": "",
         "Domain": ""
-    }
+    },
+    "GTMID": ""
```
[the remark this answers is not in the document]
Comment (Ashish15251) [reply UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/appsettings.Development.json]: Changes Done

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Common/Extensions/DateTimeExtensions.cs [hunk 2]
```diff
@@ -6,7 +6,7 @@ public static class DateTimeExtensions
     {
         public static string ToRnwDateFormat(this DateTime? dateTime)
         {
-            return dateTime == null ? string.Empty: dateTime?.ToString("ddhhmm UTC MMM yy", DateTimeFormatInfo.InvariantInfo);
+            return dateTime == null ? string.Empty: dateTime?.ToString("ddHHmm UTC MMM yy", DateTimeFormatInfo.InvariantInfo);
         }
```
[the remark this answers is not in the document]
Comment (Ashish15251) [reply UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Common/Extensions/DateTimeExtensions.cs:10]: Changes Done

**Comment by Ashish15251** [comment]
Automation test cases not required.

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/appsettings.json [hunk 3]
```diff
@@ -47,5 +47,6 @@
         "RedirectBaseUrl": ""
     },
     "APPINSIGHTS_INSTRUMENTATIONKEY": "",
-    "AllowedHosts": "*"
+    "AllowedHosts": "*",
+    "GTMID": ""
```
Comment (barrie-cooper) [inline UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/appsettings.json:51]: Could we have this as `GoogleTagManagerID` please:
```suggestion
    "GoogleTagManagerID": ""
```

**Review by usha13529 — APPROVED** [review]
Thanks @Ashish15251. Looks good to me.

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Views/Home/Index.cshtml [hunk 4]
```diff
@@ -44,9 +45,35 @@
         right: 25px;
     }
 </style>
+<!-- Google Tag Manager -->
+<script>
+    (function (w, d, s, l, i) {
```
[the remark this answers is not in the document]
Comment (Ashish15251) [reply UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Views/Home/Index.cshtml]: Done

**Pull request merged**